dotCMS/src/integration-test/java/com/dotcms/datagen/AppDescriptorDataGen.java

```diff
@@ -0,0 +1,166 @@
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return object;
+    }
+
+    private InputStream persistDescriptor(final AppDescriptor object) throws IOException {
+        String basePath = System.getProperty("java.io.tmpdir");
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath, fileName);
+        ymlMapper.writeValue(file, object);
+        return Files.newInputStream(Paths.get(file.getPath()));
+    }
+
+    public InputStream nextPersistedDescriptor() throws IOException {
+        return persistDescriptor(next());
+    }
+
+    public void destroy() throws IOException {
+        String basePath = System.getProperty("java.io.tmpdir");
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath, fileName);
+        Files.delete(Paths.get(file.getPath()));
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen key(final String key) {
+        this.key = key;
+        return this;
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen name(final String name) {
+        this.name = name;
+        return this;
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen description(final String description) {
+        this.description = description;
+        return this;
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen fileName(final String fileName) {
+        this.fileName = fileName;
+        return this;
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen allowExtraParameters(final boolean allowExtraParameters) {
+        this.allowExtraParameters = allowExtraParameters;
+        return this;
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen iconUrl(final String iconUrl) {
+        this.iconUrl = iconUrl;
+        return this;
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen param(final String name, final ParamDescriptor paramDescriptor){
+        builder.put(name, paramDescriptor);
+        return this;
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen param(final String name, boolean hidden, final Type type,
+            final String label, final String hint, final boolean required) {
+       return param(name, ParamDescriptor.newParam(StringPool.BLANK, hidden, type, label, hint, required));
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen boolParam(final String name, boolean hidden, final boolean defValue, final String label,
+            final String hint, final boolean required) {
+        return  param( name,  hidden, Type.BOOL, label, hint, required);
+    }
+
+    @SuppressWarnings("unused")
+    public AppDescriptorDataGen boolParam(final String name, boolean hidden, final boolean defValue,  final boolean required) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'hidden' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)